### PR TITLE
FIX

### PR DESCRIPTION
**Pull request by Ollieollieolllie**
The complete fix for ECM script
changed 30mm to the newer ballistics computer - ballisticcomputer = 16
Re-added ADA database - fza_ah64_ada_units list
fix for the TSD/FCR Radar Icons (shot at Still does not work)
Re-added missile datebase - fza_ah64_mis_rf/ir
fixed the position issue for the gunner seat engine 1 fire button - main p3d
Fixed the broken shadow on the FCR -longbow.p3d
disabled components of targetupdating function that broke ase page and fixed error codes
changes location of cmimmunity value so the radar hellfire stops breaking in LO trajectory

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf [hunk 1]
```diff
@@ -53,7 +53,7 @@ private _targetsToDraw = fza_ah64_targetlist apply {
 		_targetPriority = 1;
 	};
 
-	private _targIcon = format ["\fza_ah64_US\tex\ICONS\ah64_%1%2.paa", _targetType, _targetModifier];
+	private _targIcon = format [_targetType, _targetModifier];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf:56]: What were you trying to fix?
This breaks highlighting the selected target and different colours for PFZ colours. 

> fcr/tds icon sorter didn't work and leaves them at default 

I am assuming this is that, we can work on a solution to it together if you want?

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRTSDDraw.sqf [hunk 2]
```diff
@@ -86,7 +86,7 @@ if (_heli getVariable "fza_ah64_tsdmode" == "atk") then {
 			_targetPriority = 1;
 		};
 
-		private _targIcon = format ["\fza_ah64_US\tex\ICONS\ah64_%1%2.paa", _targetType, _targetModifier];
+		private _targIcon = format [_targetType, _targetModifier];
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRTSDDraw.sqf:89]: See above comment on `fn_mpdRFCRDraw.sqf`

**Review by mattysmith22 — CHANGES_REQUESTED** [review]
I like some of the changes, thank you for reintegrating CMSmk2.sqf. However, I have some issues but have provided comments explaining where there are some areas for improvement.

Also, do you mind changing the title of the PR to something more informative like "CBA compact fix set", and copy the commit changes into the description too. It will just help me when it comes to looking at these details in the future.

**Comment by SachaOropeza** [comment]
Are those ready for merge @mattysmith22 ? 🤔😉

**Comment by SachaOropeza** [comment]
@Ollieollieolllie Please be more explicit on your commit titles & description. I'm getting lost. 😅

**Comment by mattysmith22** [comment]
I don't know, haven't had time to fix. I'm quite busy today and tomorrow, I may have time Sunday though!

I agree with Sacha though, commit messages are amazing to work out what you intended to change and allows us to compare that to what has been changed. "Fix" unfortunately isn't descriptive enough.

@SachaOropeza I would recommend [squashing when you merge this commit](https://github.blog/2016-04-01-squash-your-commits/) to keep history clean and descriptive

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf [hunk 3]
```diff
@@ -21,12 +21,12 @@ if (_heli getVariable "fza_ah64_agmode" == 1) then {
 };
 
 private _targetsToDraw = fza_ah64_targetlist apply {
-	private _targetType = "gen";
+	private _targicon = "gen";
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf:24]: Out of curiosity, what was the thought process behind changing this name here? I thought `_targetType` was more descriptive in what is being stored and keeps the same variable name as in `fza_fnc_mpdRTSDDraw`. `_targicon` is also not in camel case and collides with another use of `_targIcon` below which while not really mattering could cause confusion for other readers.

If there is a reason though, that's fine!

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingUpdate.sqf [hunk 4]
```diff
@@ -19,20 +19,20 @@ Author:
 	Unknown
 ---------------------------------------------------------------------------- */
 params ["_heli"];
-
-fza_ah64_asethreats = fza_ah64_targetlist select { alive _x && _x call fza_fnc_targetIsADA && [side _x, side _heli] call BIS_fnc_sideIsEnemy};
+//Disabling this line below fixes the ASe page & The contents below are already beign done on page_ase
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/targeting/fn_targetingUpdate.sqf:22]: I'm going to do some testing of my own in-game

**Review by mattysmith22 — COMMENTED** [review]
Much better, it is looking a lot cleaner!

I am not explicitly approving yet because I want to do some actual checks in-game before I say that it is fine but just a read through of the code shows significant improvements! Your commit messages since Sacha's comment are also much better.

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf [hunk 5]
```diff
@@ -1,132 +1,37 @@
...
-		_bank = _pbvar select 1;
-		_bank = _bank - (random _chance1);
-		_pitch = _pitch - (random _chance1);
-		[_missile, _pitch, _bank] call fza_ah64_setpb;
-		sleep 0.1;
-	};
-};
+if(!(_munition isKindOf "missileBase") || !(isengineon _ac)) exitwith {};
 
 if (!(isNil "fza_ah64_noscripts") || !(isengineon _ac) || !(alive _ac) || !(_munition isKindOf "MissileBase") || !(player in _ac)) exitwith {};
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf]: SORTED

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/fcr_longbow.sqf [hunk 6]
```diff
@@ -12,7 +12,10 @@ _adaunit = false;
 _datalinkArray = [];
 
 if (isNil "fza_ah64_ada_units") then {
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/fcr_longbow.sqf]: SORTED

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssDraw.sqf [hunk 7]
```diff
@@ -867,6 +867,12 @@ if ((diag_tickTime % 2) < 1) then {
     _safemessage = "";
 };
 
+//auto hover detector//
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/ihadss/fn_ihadssDraw.sqf]: SORTED

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf [hunk 8]
```diff
@@ -1,132 +1,37 @@
...
-
 _posac = getpos _ac;
 _poshostile = getpos _missile;
 _range = _poshostile distance _posac;
 _highlow = "High";
 
-if(typeOf _ac == "fza_ah64d_b2e") then
+_fza_ah64_incominghandled = _hostile getVariable ["fza_ah64_shotCounter", 0];
+
+/*5 / 2 => 2
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf:23]: SORTED

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf [hunk 9]
```diff
@@ -255,41 +128,66 @@ if (_theta > 315 && _theta < 346) then
...
+	_chance1 = 200;
+	_rand = 5;
+
+	if(typeof _missile in fza_ah64_mis_ir) then {_rand = 6;};
+	if(typeof _missile in fza_ah64_mis_rf) then {_rand = 4;};
+	if (floor random 10 < _rand) exitwith {};
+	waitUntil {_missile distance _ac < 200};	
+	while {(alive _missile) && (alive _ac)} do
+	{
+		_reldir = ((getposasl _ac select 0) - (getposasl _missile select 0)) atan2 ((getposasl _ac select 1) - (getposasl _missile select 1));
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf]: SORTED

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf [hunk 10]
```diff
@@ -255,41 +128,66 @@ if (_theta > 315 && _theta < 346) then
 	_clockaud = "fza_ah64_bt_11oclock";
 };
 
-_curvupmis = vectorup _missile;
-_curvdirmis = vectordir _missile;
-
-_reldir = ((getposasl _ac select 0) - (getposasl _missile select 0)) atan2((getposasl _ac select 1) - (getposasl _missile select 1));
-if (_reldir < 0) then {
-    _reldir = _reldir + 360;
+_spoken = 1;
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf]: SORTED

[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf]: This was from when I thought the script is self was rerunning

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf [hunk 11]
```diff
@@ -255,41 +128,66 @@ if (_theta > 315 && _theta < 346) then
...
+_read = 1;
+if(!(_hostile in fza_ah64_threatfiring)) then {fza_ah64_threatfiring = fza_ah64_threatfiring + [_hostile];};
+if (_ac getVariable "fza_ah64_aseautopage" == 2) then {
+        [_ac, 1, "ase"] call fza_fnc_mpdSetDisplay;
+    };
+if (_read == 1 && _range < 8000) then
```
[the remark this answers is not in the document]
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/ecm/CMSmk2.sqf]: SORTED

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf (hunk 1 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf]: Fixed

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRTSDDraw.sqf (hunk 2 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRTSDDraw.sqf]: fixed

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf (hunk 3 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/mpd/fn_mpdRFCRDraw.sqf]: Changed

**Pull request merged**